### PR TITLE
[RFR] Composedof attribute 2

### PR DESCRIPTION
**Pull request by ThieryMichel**
Admin part
capture du 2017-02-28 16 14 14

### CODE REVIEW

File: src/app/e2e/admin/publication.spec.js [hunk 1]
```diff
@@ -38,11 +38,6 @@ describe('Admin', () => {
             it('should display only uri empty column', async () => {
                 const th = await driver.findElement(By.css('.publication-preview th'));
                 driver.wait(until.elementTextIs(th, 'uri'), DEFAULT_WAIT_TIMEOUT);
-                const tds = await driver.findElements(By.css('.publication-preview tr td:first-child'));
```
Comment (djhi) [inline src/app/e2e/admin/publication.spec.js]: why removing this ? We're not testing for emptiness anymore

File: src/app/js/admin/fields/ClearComposedOf.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import FlatButton from 'material-ui/FlatButton';
+
+import { polyglot as polyglotPropTypes } from '../../propTypes';
+import { clearComposedOf } from './';
+
+export const ClearComposedOfComponent = ({ p: polyglot, onClick }) => (
+    <FlatButton onClick={onClick} label={polyglot.t('remove')} />
+);
+
+
```
Comment (djhi) [inline src/app/js/admin/fields/ClearComposedOf.js]: extra line

File: src/app/js/admin/fields/AddComposedOfField.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+} from './';
+
+export const AddComposedOfComponent = ({ onAdd, onRemove }) => (
+    <div>
+        <FlatButton onClick={onAdd} label="+" />
+        <FlatButton onClick={onRemove} label="-" />
+    </div>
+);
+
+
```
Comment (djhi) [inline src/app/js/admin/fields/AddComposedOfField.js]: extra line

File: src/app/js/admin/fields/AddComposedOf.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import FlatButton from 'material-ui/FlatButton';
+
+import { polyglot as polyglotPropTypes } from '../../propTypes';
+import { addComposedOf } from './';
+
+export const AddComposedOfComponent = ({ p: polyglot, onClick }) => (
+    <FlatButton onClick={onClick} label={polyglot.t('add')} />
+);
+
+
```
Comment (djhi) [inline src/app/js/admin/fields/AddComposedOf.js]: extra line

File: src/app/js/admin/publicationPreview/PublicationExcerptLineCol.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+const mapStateToProps = (state, { field, line }) => {
+    const getLineCol = fromFields.getLineColGetter(state, field);
+
+    return {
+        value: getLineCol(line),
+    };
+};
+
```
Comment (djhi) [inline src/app/js/admin/publicationPreview/PublicationExcerptLineCol.js:45]: extra line

**Review by djhi — CHANGES_REQUESTED** [review]
Great! Can you add a screenshot ?

File: src/app/js/admin/publicationPreview/PublicationExcerptLineCol.js (hunk 5 above)
Comment (djhi) [reply src/app/js/admin/publicationPreview/PublicationExcerptLineCol.js:45]: missed this one ?

Comment (ThieryMichel) [reply src/app/js/admin/publicationPreview/PublicationExcerptLineCol.js:45]: done

Comment (ThieryMichel) [reply src/app/js/admin/publicationPreview/PublicationExcerptLineCol.js:45]: done

File: src/app/e2e/admin/publication.spec.js (hunk 1 above)
Comment (djhi) [reply src/app/e2e/admin/publication.spec.js]: Any answer on this one ?

**Review by djhi — APPROVED** [review]

**Pull request merged**